### PR TITLE
TASK_LOST tasks should not be relayed to Consul

### PR DESCRIPTION
**Pull request by smn**
Upstream nginx will result in proxying traffic to tasks marked as lost.

### CODE REVIEW

**Comment by smn** [comment]
@miltontony @JayH5 ready for review

**Comment by jerith** [comment]
If you add a failing test that you're going to fix in the next commit, http://doc.pytest.org/en/latest/skipping.html#mark-a-test-function-as-expected-to-fail is useful. :-)

File: consular/tests/test_main.py [hunk 1]
```diff
@@ -645,6 +645,60 @@ def test_sync_app_tasks_grouped(self):
         yield d
 
     @inlineCallbacks
+    def test_sync_app_tasks_task_lost(self):
+        """
+        When syncing an app with a task, Consul is updated with a service entry
+        for the task.
+        """
```
Comment (jerith) [inline consular/tests/test_main.py:660]: I think this docstring needs updating.

File: consular/tests/test_main.py [hunk 2]
```diff
@@ -645,6 +652,60 @@ def test_sync_app_tasks_grouped(self):
...
```
Comment (JayH5) [inline consular/tests/test_main.py:660]: Better description please.

File: setup.cfg [hunk 3]
```diff
@@ -1,2 +1,2 @@
-[pytest]
+[tool:pytest]
```
Comment (JayH5) [inline setup.cfg:1]: Maybe pin pytest >= 3.0.0 for this change

File: consular/tests/test_main.py [hunk 4]
```diff
@@ -1110,6 +1111,95 @@ def test_purge_dead_services(self):
         yield d
 
     @inlineCallbacks
+    def test_purge_task_lost_services(self):
+        d = self.consular.purge_dead_services()
```
Comment (jerith) [inline consular/tests/test_main.py:1145]: This test needs a docstring.

File: consular/tests/test_main.py [hunk 5]
```diff
@@ -1110,6 +1111,95 @@ def test_purge_dead_services(self):
...
+                    "Tags": [
+                        "consular-reg-id=test",
+                        "consular-app-id=/testinggroup/someid1",
+                    ],
+                }
+            }))
+        )
+
+        # Expecting a request for the tasks for a given app, returning
+        # 1 of which only has the TASK_RUNNING state.
```
Comment (jerith) [inline consular/tests/test_main.py]: This comment doesn't make much sense. Maybe "returning 2 of which only 1 has the TASK_RUNNING state"?

**Comment by jerith** [comment]
:+1:

**Comment by JayH5** [comment]
:+1: when Travis

**Pull request merged**